delfin/exporter/prometheus/exporter_server.py

```diff
@@ -11,34 +11,48 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
+import glob
+import os
 
 from flask import Flask
 from oslo_config import cfg
 import sys
+from oslo_log import log
+
+LOG = log.getLogger(__name__)
 
 app = Flask(__name__)
 
 grp = cfg.OptGroup('PROMETHEUS_EXPORTER')
-
+METRICS_CACHE_DIR = '/var/lib/delfin/metrics'
 prometheus_opts = [
     cfg.StrOpt('metric_server_ip', default='0.0.0.0',
                help='The exporter server host  ip'),
     cfg.IntOpt('metric_server_port', default=8195,
                help='The exporter server port'),
-    cfg.StrOpt('metrics_cache_file', default='/var/lib/delfin/delfin_exporter'
-                                             '.txt',
-               help='The temp cache file used for persisting metrics'),
+    cfg.StrOpt('metrics_dir', default=METRICS_CACHE_DIR,
+
+               help='The temp directory to keep incoming metrics'),
 ]
 cfg.CONF.register_opts(prometheus_opts, group=grp)
 cfg.CONF(sys.argv[1:])
 
 
 @app.route("/metrics", methods=['GET'])
 def getfile():
-    with open(cfg.CONF.PROMETHEUS_EXPORTER.metrics_cache_file, "r+") as f:
-        data = f.read()
-        f.truncate(0)
+    try:
+        os.chdir(cfg.CONF.PROMETHEUS_EXPORTER.metrics_dir)
+    except OSError as e:
+        LOG.error('Error opening metrics folder')
+        raise Exception(e)
+    data = ''
+    for file in glob.glob("*.prom"):
+        file_name = cfg.CONF.PROMETHEUS_EXPORTER.metrics_dir + '/' + file
```

Comment:
Can we use os.path.join() here. This will make this helpful for platform independent

Reply:
Done thanks